pkg/cri/server/podsandbox/recover.go

```diff
@@ -103,37 +96,48 @@ func (c *Controller) RecoverContainer(ctx context.Context, cntr containerd.Conta
 			status.State = sandboxstore.StateNotReady
 		} else {
 			if taskStatus.Status == containerd.Running {
-				// Wait for the task for sandbox monitor.
-				// wait is a long running background request, no timeout needed.
+				status.State = sandboxstore.StateReady
```

Comment:
why are we setting ready if there is still a window for failure to wait for the container task (see return below)

Reply:
@abel-von ...

Reply:
Sorry for the late reply, I think you are right, I have to set the status to NotReady before return, I will submit another PR to do it

Reply:
@mikebrow  #9644